libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/ReloadingComponent.kt

```diff
@@ -26,10 +29,12 @@ internal class ReloadingComponent<T>(
 
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
-        flow.observe {
```

Comment:
This observe function is part of `UIComponent`. Why not make the `Dispatchers.Main.immediate` part of the `observe` extension function? At first glance, this looks like something that will be faster for everything.

A concern is that this could also uncover some issues relying on the coroutine scope switching. There were some issues we had early on that were impacted by it. So it will require testing just about everything in drop in ui.

Reply:
> This observe function is part of UIComponent. Why not make the Dispatchers.Main.immediate part of the observe extension function? 

I thought that I only needed it here so why change it for everything else?

> A concern is that this could also uncover some issues relying on the coroutine scope switching.

Yes, but the alternative way is to no rely on flows here at all (or somehow guarantee the order in which the first item is delivered ourselves). It's also an option, but the current solution seemed quite organic with the current approach so I demonstrated it.

If you're worried about it, let's discuss the other option.

Reply:
> I thought that I only needed it here so why change it for everything else?

Curious if you looked into it. Using it is crazy faster

Before in seconds
```
observe time 0.084311518
observe time 0.070879747
observe time 0.07200782
observe time 0.073052299
```

Before in seconds
```
observe time 6.95313E-4
observe time 3.73802E-4
observe time 2.3932300000000003E-4
observe time 3.29115E-4
```

It essentially speeds it up by a factor of 100

Reply:
Here's the quick perf test code I added for reference. Notice the only difference is the `Dispatchers.Main.immediate`. Also, after smoke testing a few things - all of dropinui appears to behave the same.

Before perf timing
``` kotlin
    protected inline fun <T> Flow<T>.observe(crossinline action: suspend (value: T) -> Unit) {
        val startTime = System.nanoTime()
        var firstLaunchTime: Long? = null
        coroutineScope.launch {
            collect {
                if (firstLaunchTime == null) {
                    firstLaunchTime = System.nanoTime()
                    val launchTime = (firstLaunchTime!! - startTime) * 	1.0E-9
                    logI("observe time $launchTime", "kyle_debug")
                }
                action(it)
            }
        }
    }
```

After
```
    protected inline fun <T> Flow<T>.observe(crossinline action: suspend (value: T) -> Unit) {
        val startTime = System.nanoTime()
        var firstLaunchTime: Long? = null
        coroutineScope.launch(Dispatchers.Main.immediate) {
            collect {
                if (firstLaunchTime == null) {
                    firstLaunchTime = System.nanoTime()
                    val launchTime = (firstLaunchTime!! - startTime) * 	1.0E-9
                    logI("observe time $launchTime", "kyle_debug")
                }
                action(it)
            }
        }
    }
```

Reply:
It makes sense that it's faster here but don't forget that it's the main scope. Won't we make something else slower because we want the thread here and now? "Something else" may be very important as it's also on the main thread.

Reply:
By the way, `LifecycleOwner`s and `ViewModel`s use `CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)` instead of `MainScope`. I think we should do the same in `UIComponent`.

Reply:
Done.